spec/ics-23-vector-commitments/README.md

```diff
@@ -0,0 +1,250 @@
+
+```coffeescript
+remove(CommitmentState state, Key key) -> CommitmentState
+```
+
+#### Proof generation
+
+The `createMembershipProof` function generates a proof that a particular key has been set to a particular value in an commitment.
+
+```coffeescript
+createMembershipProof(CommitmentState state, Key key, Value value) -> CommitmentProof
+```
+
+The `createNonMembershipProof` function generates a proof that a key has not been set to any value in an commitment.
+
+```coffeescript
+createNonMembersipProof(CommitmentState state, Key key) -> CommitmentProof
+```
+
+#### Proof verification
+
+The `verifyMembership` function verifies a proof that a key has been set to a particular value in an commitment.
+
+```coffeescript
+verifyMembership(CommitmentRoot root, CommitmentProof proof, Key key, Value value) -> boolean
+```
+
+The `verifyNonMembership` function verifies a proof that a key has not been set to any value in an commitment.
+
+```coffeescript
+verifyNonMembership(CommitmentRoot root, CommitmentProof proof, Key key) -> boolean
+```
+
+### Optional functions
+
+An commitment construction MAY provide the following functions:
+
+The `batchVerifyMembership` function verifies a proof that many keys have been set to specific values in an commitment.
+
+```coffeescript
+batchVerifyMembership(CommitmentRoot root, CommitmentProof proof, Map<Key, Value> items) -> boolean
+```
+
+The `batchVerifyNonMembership` function verifies a proof that many keys have not been set to any value in an commitment.
+
+```coffeescript
+batchVerifyNonMembership(CommitmentRoot root, CommitmentProof proof, Set<Key> keys) -> boolean
+```
+
+If defined, these functions MUST be computationally equivalent to the conjunctive union of `verifyMembership` and `verifyNonMembership` respectively (`proof` may vary):
+
+```coffeescript
+batchVerifyMembership(root, proof, items) ==
+  verifyMembership(root, proof, items[0].Key, items[0].Value) &&
+  verifyMembership(root, proof, items[1].Key, items[1].Value) && ...
+```
+
+```coffeescript
+batchVerifyMembership(root, proof, keys) ==
+  verifyNonMembership(root, proof, keys[0]) &&
+  verifyNonMembership(root, proof, keys[1]) && ...
+```
+
+If batch verification is possible and more efficient than individual verification of one proof per element, an commitment construction SHOULD define batch verification functions.
+
+### Properties
+
+Commitments must be *complete*, *sound*, and *position binding*. These properties are defined with respect to a security parameter `λ`, which must be agreed upon by the manager, prover, and verifier (and often will be constant for the commitment algorithm).
+
+#### Completeness
+
+Commitment proofs must be *complete*: key => value mappings which have been added to the commitment can always be proved to have been included, and keys which have not been included can always be proved to have been excluded, except with probability negligible in `λ`.
```

Comment:
"except with probability negligible in `λ`." Should this be slightly more formal once in this document or in the glossary document. E.g. stating more formally that the probability of the opposite can be expressed as a function which is negligible in λ. Or point to some definition somewhere?

Reply:
Defined more clearly and linked to Wikipedia's definition (which is clear)